devices/src/virtio/vsock/epoll_handler.rs

```diff
@@ -55,14 +79,68 @@ where
     /// have pending.
     ///
     fn process_rx(&mut self) {
-        error!("vsock: EpollHandler::process_rx() NYI");
+        debug!("vsock: epoll_handler::process_rx()");
+
+        let mut raise_irq = false;
+
+        while let Some(head) = self.rxvq.iter(&self.mem).next() {
+            let used_len = match VsockPacket::from_rx_virtq_head(&head) {
+                Ok(mut pkt) => {
+                    if self.backend.recv_pkt(&mut pkt).is_ok() {
+                        pkt.hdr.as_slice().len() as u32 + pkt.hdr.len
+                    } else {
+                        // We are using a consuming iterator over the virtio buffers, so, if we can't
```

Comment:
I think this could lead to errors: either you lose a packet, which vsock doesn't like or you get stuck processing the same packet forever.

Reply:
I'm not sure I understand how that could happen. Can you give me some more details, please?

Reply:
I think what @petreeftime is referring to is that since you are undoing the `pop` when the backend fails to `recv_pkt` , you can get stuck trying to get the faulty packet through. Is that possible?